snakemake/rules.py

```diff
@@ -1124,20 +1118,40 @@ def expand_group(self, wildcards):
         else:
             return self.group
 
-    def expand_conda_env(self, wildcards):
-        try:
-            conda_env = (
-                self.conda_env.apply_wildcards(wildcards, self)
-                if self.conda_env
-                else None
-            )
-        except WildcardError as e:
-            raise WildcardError(
-                "Wildcards in conda environment file cannot be "
-                "determined from output files:",
-                str(e),
-                rule=self,
+    def expand_conda_env(self, wildcards, params=None, input=None):
+        from snakemake.deployment.conda import (
+            is_conda_env_file,
+            CondaEnvFileSpec,
+            CondaEnvNameSpec,
+        )
+
+        if callable(self._conda_env):
+            conda_env, _ = self.apply_input_function(
+                self._conda_env, wildcards=wildcards, params=params, input=input
             )
+        else:
```

Comment:
This does not need to be in an else, but instead done always, also if it was a callable.